### PR TITLE
feat(sdk): Add interface to support basic auth and kerberos SPNEGO reverse proxy

### PR DESCRIPTION
**Pull request by andres-de-castro**
Added username and password support for basic auth proxies.

basic auth usage: 

from urrllib import make_headers

basic_auth_headers = make_headers(basic_auth=f"{proxy_username}:{proxy_password}")

client = kfp.Client(HOST_NAME, proxy=YOUR_PROXY_HOST, proxy_headers=basic_auth_headers)

kerberos usage 

from requests_kerberos import HTTPKerberosAuth

kerb_auth =  HTTPKerberosAuth(mutual_authentication="MUTUAL", force_preemptive=True)
auth_header = kerb_auth.generate_request_header(None, HOSTNAME, True)
client = kfp.Client(HOST_NAME, proxy=YOUR_PROXY_HOST, proxy_headers={"Proxy-Authorization": auth_header})

### CODE REVIEW

**Comment by andres-de-castro** [comment]
@googlebot I signed it!

File: sdk/python/kfp/_client.py [hunk 1]
```diff
@@ -33,6 +33,8 @@
 
 from kfp._auth import get_auth_token, get_gcp_access_token
 
+from urllib3 import make_headers
```
Comment (casassg) [inline sdk/python/kfp/_client.py:36]: This seems not used?

**Comment by Bobgy** [comment]
@andres-de-castro we are moving towards building an interface for auth in https://github.com/kubeflow/pipelines/pull/5287.

Can we improve that interface if it doesn't allow your use case for now? This will make sure we can organize auth methods in a maintainable and pluggable way.

**Comment by rimolive** [comment]
Closing this PR. No activity for more than a year.

/close

**Pull request closed without merging**